File: src-tauri/src/game/insights.rs
use serde::{Deserialize, Serialize};
use super::state::GameState;
use super::competitors::{get_most_threatening_competitor, get_shipping_velocity_ratio};

/// Educational insight about player's decisions and game state
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WeeklyInsight {
    pub category: InsightCategory,
    pub title: String,
    pub observation: String,
    pub insight: String,
    pub action_suggestion: String,
    pub severity: InsightSeverity,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum InsightCategory {
    Morale,
    TechnicalDebt,
    Runway,
    Growth,
    CustomerSatisfaction,
    Velocity,
    Burnout,
    Competition,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum InsightSeverity {
    Info,      // Educational, no urgency
    Warning,   // Attention needed
    Critical,  // Immediate action required
}

/// Generate insights by comparing previous and current state
pub fn generate_weekly_insights(prev_state: &GameState, curr_state: &GameState) -> Vec<WeeklyInsight> {
    let mut insights = Vec::new();

    // 1. Morale Check
    if curr_state.morale < prev_state.morale - 10.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Morale,
            title: "Team Morale Declining".to_string(),
            observation: format!(
                "Morale dropped from {:.0}% to {:.0}% this week",
                prev_state.morale, curr_state.morale
            ),
            insight: "Burned out teams make poor decisions and ship slower. Morale isn't 'soft' - it's a leading indicator of productivity and retention. Low morale leads to key people leaving, taking institutional knowledge with them.".to_string(),
            action_suggestion: "Consider: Take Break action, reduce workload, celebrate wins, or slow down shipping pace".to_string(),
            severity: if curr_state.morale < 40.0 {
                InsightSeverity::Critical
            } else if curr_state.morale < 60.0 {
                InsightSeverity::Warning
            } else {
                InsightSeverity::Info
            },
        });
    }

    // 2. Morale Sustained High
    if curr_state.morale > 75.0 && prev_state.morale > 75.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Morale,
            title: "Strong Team Culture".to_string(),
            observation: format!("Team morale has been consistently high at {:.0}%", curr_state.morale),
            insight: "Sustainable pace wins. High morale compounds: happy teams ship better code faster, make better decisions, and attract better talent. This is a competitive advantage.".to_string(),
            action_suggestion: "Keep doing what you're doing. Document what's working so you can maintain it as you scale.".to_string(),
            severity: InsightSeverity::Info,
        });
    }

    // 3. Technical Debt Warning
    if curr_state.tech_debt > 60.0 && prev_state.tech_debt < 60.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::TechnicalDebt,
            title: "Technical Debt Accumulating".to_string(),
            observation: format!("Tech debt crossed 60% threshold (now at {:.0}%)", curr_state.tech_debt),
            insight: "High tech debt is like driving with the parking brake on. Every feature takes longer. Every change risks breaking something. The 'interest payments' on tech debt compound - it gets exponentially harder to fix the longer you wait.".to_string(),
            action_suggestion: "Ship fewer features this week and focus on quality. Refactor critical paths. Future you will thank present you. Consider 'Polish' quality for next features.".to_string(),
            severity: if curr_state.tech_debt > 80.0 {
                InsightSeverity::Critical
            } else {
                InsightSeverity::Warning
            },
        });
    }

    // 4. Tech Debt Under Control
    if curr_state.tech_debt < 30.0 && curr_state.velocity > 0.8 {
        insights.push(WeeklyInsight {
            category: InsightCategory::TechnicalDebt,
            title: "Engineering Excellence".to_string(),
            observation: format!("Low tech debt ({:.0}%) with strong velocity ({:.1}x)", curr_state.tech_debt, curr_state.velocity),
            insight: "Quality and speed aren't opposites. Clean codebases enable speed. You can ship features faster because you're not fighting technical debt. This is how great engineering teams win.".to_string(),
            action_suggestion: "Maintain this discipline. Make it part of your culture. Every 'quick hack' has a compounding cost.".to_string(),
            severity: InsightSeverity::Info,
        });
    }

    // 5. Runway Warning
    if curr_state.runway_months < 6.0 {
        let urgency_level = if curr_state.runway_months < 3.0 {
            InsightSeverity::Critical
        } else {
            InsightSeverity::Warning
        };

        insights.push(WeeklyInsight {
            category: InsightCategory::Runway,
            title: "Runway Running Low".to_string(),
            observation: format!("Only {:.1} months of runway remaining", curr_state.runway_months),
            insight: "Short runway forces desperate decisions. You need runway to think clearly, negotiate fairly, and build correctly. Desperation is expensive - you'll accept bad deals, cut corners, and make fear-based choices. Runway is optionality.".to_string(),
            action_suggestion: "Either raise money, cut burn significantly, or increase revenue. Start fundraising NOW if that's your path - it takes 3-6 months. Consider hard choices before you're forced to make them.".to_string(),
            severity: urgency_level,
        });
    }

    // 6. Strong Runway
    if curr_state.runway_months > 18.0 && curr_state.burn > 0.0 {
        let burn_efficiency = curr_state.mrr / curr_state.burn;
        if burn_efficiency > 0.5 {
            insights.push(WeeklyInsight {
                category: InsightCategory::Runway,
                title: "Financial Discipline Pays Off".to_string(),
                observation: format!("{:.1} months runway with healthy burn efficiency", curr_state.runway_months),
                insight: "Runway is freedom. Cash gives you the power to make good decisions from strength, not desperation. You can say no to bad deals, wait for the right opportunities, and invest in quality. This is a massive strategic advantage.".to_string(),
                action_suggestion: "Leverage this position. You can afford to be patient and strategic. Consider: quality hiring, deeper customer research, or technical investments that pay long-term dividends.".to_string(),
                severity: InsightSeverity::Info,
            });
        }
    }

    // 7. Growth Stagnation
    if curr_state.wau_growth_rate < 2.0 && curr_state.week > 8 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Growth,
            title: "Growth Stagnating".to_string(),
            observation: format!("WAU growth at only {:.1}% per week", curr_state.wau_growth_rate),
            insight: "Stagnant growth is a signal. Either: (1) You haven't found product-market fit yet, (2) You're not reaching your market effectively, or (3) Churn is eating your growth. Time to diagnose which one.".to_string(),
            action_suggestion: "Talk to customers. What are they getting value from? What would make them tell friends? Consider: better onboarding, key feature improvements, or finding a different channel.".to_string(),
            severity: InsightSeverity::Warning,
        });
    }

    // 8. Strong Growth with High Churn
    if curr_state.wau_growth_rate > 10.0 && curr_state.churn_rate > 10.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::CustomerSatisfaction,
            title: "Leaky Bucket".to_string(),
            observation: format!("Strong growth ({:.1}%) but high churn ({:.1}%)", curr_state.wau_growth_rate, curr_state.churn_rate),
            insight: "You're acquiring customers but not keeping them - the classic 'leaky bucket'. Every lost customer took resources to acquire and might spread negative word of mouth. Keeping customers happy is always cheaper than acquiring new ones.".to_string(),
            action_suggestion: "Interview churned customers. Find the common pattern. Often it's: onboarding issues, missing key feature, or misaligned expectations. Fix retention before scaling acquisition.".to_string(),
            severity: InsightSeverity::Warning,
        });
    }

    // 9. Velocity Degradation
    if curr_state.velocity < 0.7 && prev_state.velocity >= 0.7 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Velocity,
            title: "Shipping Velocity Declining".to_string(),
            observation: format!("Velocity dropped to {:.1}x (was {:.1}x)", curr_state.velocity, prev_state.velocity),
            insight: "Declining velocity is usually caused by: (1) Tech debt making every change expensive, (2) Team morale/energy issues, or (3) Growing complexity without enough refactoring. The longer you wait, the worse it gets.".to_string(),
            action_suggestion: "Identify the bottleneck. If it's tech debt, allocate time for refactoring. If it's morale, take a break. If it's complexity, simplify ruthlessly. Velocity is a leading indicator of future productivity.".to_string(),
            severity: InsightSeverity::Warning,
        });
    }

    // 10. Burnout Risk (multiple indicators)
    let weeks_since_break = check_weeks_since_break(curr_state);
    if weeks_since_break > 8 && curr_state.morale < 70.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Burnout,
            title: "Burnout Risk".to_string(),
            observation: format!("{} weeks without a break, morale at {:.0}%", weeks_since_break, curr_state.morale),
            insight: "Burnout doesn't happen overnight - it accumulates. Warning signs: tired in meetings, declining code quality, cynicism, dread on Monday mornings. Once you're burned out, recovery takes months. Prevention is easier than cure.".to_string(),
            action_suggestion: "Take a proper break this week. Not a weekend - a real vacation. Your business needs you healthy and clear-headed more than it needs you grinding 80-hour weeks.".to_string(),
            severity: InsightSeverity::Critical,
        });
    }

    // 11. Customer Love Achievement
    if curr_state.nps > 70.0 && curr_state.wau > 500 {
        insights.push(WeeklyInsight {
            category: InsightCategory::CustomerSatisfaction,
            title: "Customers Love Your Product".to_string(),
            observation: format!("NPS at {:.0} with {} active users", curr_state.nps, curr_state.wau),
            insight: "High NPS with real usage is the holy grail. These customers become advocates - they sell for you, give feedback, and provide testimonials. This compounds: happy customers tell friends, which brings more customers predisposed to like you.".to_string(),
            action_suggestion: "Double down on what's working. Make it even better. Ask your champions what made them champions. Consider a referral program or case studies.".to_string(),
            severity: InsightSeverity::Info,
        });
    }

    // Competitive Intelligence Insights

    // Competitor Out-Shipping
    if let Some(competitor) = get_most_threatening_competitor(&curr_state.competitors) {
        let velocity_ratio = get_shipping_velocity_ratio(competitor, curr_state);
        if velocity_ratio > 1.5 {
            insights.push(WeeklyInsight {
                category: InsightCategory::Competition,
                title: "Competitor Out-Shipping You".to_string(),
                observation: format!("{} is shipping features {:.1}x faster than you. Their feature parity is at {:.0}% and growing.", competitor.name, velocity_ratio, competitor.feature_parity),
                insight: "Shipping velocity is a competitive weapon. Fast-moving competitors can out-innovate you, steal customers, and attract better talent. Speed compounds - they learn faster, iterate faster, and build momentum. This is how underdogs beat incumbents.".to_string(),
                action_suggestion: format!("Analyze why {} ships faster. Is it team size? Tech debt? Focus? Consider: reduce tech debt, improve velocity, or find a differentiation strategy that doesn't require matching their pace.", competitor.name),
                severity: InsightSeverity::Warning,
            });
        }
    }

    // Funding Gap
    let total_competitor_funding = curr_state.get_total_competitor_funding();
    if total_competitor_funding > curr_state.bank * 5.0 && curr_state.week > 52 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Competition,
            title: "Funding Gap Widening".to_string(),
            observation: format!("Your competitors have raised ${:.0}M combined while you have ${:.0}k in the bank. They can outspend you on hiring, marketing, and sales.", total_competitor_funding / 1_000_000.0, curr_state.bank / 1000.0),
            insight: "Capital is a competitive advantage. Well-funded competitors can afford to lose money acquiring customers, hire faster, and wait out market downturns. You need either: (1) raise money to compete, (2) find an unfair advantage that doesn't require capital, or (3) target a different market segment.".to_string(),
            action_suggestion: "Consider fundraising if you're on the VC track. Or double down on capital efficiency - find channels and strategies that don't require outspending competitors. Bootstrapped companies can win, but not by playing the same game as funded competitors.".to_string(),
            severity: InsightSeverity::Warning,
        });
    }

    // Market Share Declining
    if curr_state.player_market_share < prev_state.player_market_share - 5.0 {
        insights.push(WeeklyInsight {
            category: InsightCategory::Competition,
            title: "Losing Market Share".to_string(),
            observation: format!("Your market share dropped from {:.1}% to {:.1}% this week. Competitors are winning customers you should be winning.", prev_state.player_market_share, curr_state.player_market_share),
            insight: "Market share is a leading indicator. Losing share means: (1) competitors are executing better, (2) your product isn't differentiated enough, or (3) you're being outmarketed. In winner-take-most markets, losing share early can be fatal - network effects and brand recognition compound.".to_string(),
            action_suggestion: "Diagnose why you're losing share. Talk to customers who chose competitors. What did they offer that you don't? Is it features, pricing, brand, or distribution? Fix the root cause, not the symptoms.".to_string(),
            severity: InsightSeverity::Critical,
        });
    }

    // Competitive Advantage
    if curr_state.player_market_share > 60.0 {
        let avg_competitor_parity = curr_state.get_average_competitor_feature_parity();
        if curr_state.velocity > avg_competitor_parity / 10.0 {
            insights.push(WeeklyInsight {
                category: InsightCategory::Competition,
                title: "Competitive Moat Building".to_string(),
                observation: format!("You control {:.1}% market share and ship faster than competitors. You're building a defensible position.", curr_state.player_market_share),
                insight: "Market leadership compounds. High market share → more customers → more feedback → better product → more customers. You're in the virtuous cycle. Now the question is: how do you make this position defensible? Network effects? Switching costs? Brand? Data advantages?".to_string(),
                action_suggestion: "Don't get complacent. Dominant players get disrupted when they stop innovating. Keep shipping, keep listening to customers, and invest in moats that make it hard for competitors to catch up. Consider: platform effects, integrations, or community.".to_string(),
                severity: InsightSeverity::Info,
            });
        }
    }

    // Acquisition Opportunity
    if curr_state.bank > 200_000.0 {
        if let Some(competitor) = curr_state.competitors.iter().find(|c| !c.is_acquired && c.feature_parity < 30.0 && matches!(c.funding_stage, super::competitors::FundingStage::Bootstrapped)) {
            insights.push(WeeklyInsight {
                category: InsightCategory::Competition,
                title: "Acquisition Opportunity".to_string(),
                observation: format!("{} is struggling (feature parity: {:.0}%). They might be open to acquisition. You have the capital.", competitor.name, competitor.feature_parity),
                insight: "Acquiring competitors can be faster than building. You get their customers, team, and technology. But acquisitions are risky - cultural fit matters, integration is hard, and you might overpay. Only acquire if it accelerates your strategy, not just to eliminate competition.".to_string(),
                action_suggestion: format!("Consider reaching out to {}. But be strategic - what would you gain? Their customers? Technology? Team? Make sure the acquisition makes sense beyond just removing a competitor.", competitor.name),
                severity: InsightSeverity::Info,
            });
        }
    }

    // Limit to top 3 most important insights
    insights.sort_by_key(|i| match i.severity {
        InsightSeverity::Critical => 0,
        InsightSeverity::Warning => 1,
        InsightSeverity::Info => 2,
    });
    insights.truncate(3);

    insights
}

/// Check how many weeks since last break (placeholder - would need state tracking)
fn check_weeks_since_break(_state: &GameState) -> u32 {
    // TODO: Track this in game state
    // For now, estimate based on morale trajectory
    8
}

#[cfg(test)]
mod tests {
    use super::*;
    use crate::game::state::DifficultyMode;

    #[test]
    fn test_morale_declining_insight() {
        let mut prev_state = GameState::new(DifficultyMode::IndieBootstrap);
        prev_state.morale = 80.0;

        let mut curr_state = prev_state.clone();
        curr_state.morale = 60.0;

        let insights = generate_weekly_insights(&prev_state, &curr_state);

        assert!(!insights.is_empty());
        assert!(insights.iter().any(|i| i.category == InsightCategory::Morale));
    }

    #[test]
    fn test_runway_warning() {
        let prev_state = GameState::new(DifficultyMode::IndieBootstrap);

        let mut curr_state = prev_state.clone();
        curr_state.runway_months = 4.0;

        let insights = generate_weekly_insights(&prev_state, &curr_state);

        let runway_insight = insights.iter().find(|i| i.category == InsightCategory::Runway);
        assert!(runway_insight.is_some());
        assert_eq!(runway_insight.unwrap().severity, InsightSeverity::Warning);
    }

    #[test]
    fn test_tech_debt_warning() {
        let prev_state = GameState::new(DifficultyMode::IndieBootstrap);

        let mut curr_state = prev_state.clone();
        curr_state.tech_debt = 65.0;

        let insights = generate_weekly_insights(&prev_state, &curr_state);

        assert!(insights.iter().any(|i| i.category == InsightCategory::TechnicalDebt));
    }

    #[test]
    fn test_insights_prioritization() {
        let prev_state = GameState::new(DifficultyMode::IndieBootstrap);

        let mut curr_state = prev_state.clone();
        curr_state.morale = 30.0; // Critical
        curr_state.tech_debt = 65.0; // Warning
        curr_state.runway_months = 15.0; // Info

        let insights = generate_weekly_insights(&prev_state, &curr_state);

        // Should prioritize critical insights first
        assert!(insights.len() <= 3);
        if !insights.is_empty() {
            assert_eq!(insights[0].severity, InsightSeverity::Critical);
        }
    }
}

File: src/lib/game-engine/insights.ts
import { GameState } from '../../types/game-systems';

// Port of Rust insights generation from src-tauri/src/game/insights.rs

export interface Insight {
  category: string;
  title: string;
  observation: string;
  insight: string;
  action_suggestion: string;
  severity: 'Info' | 'Warning' | 'Critical';
  metric?: string;
  threshold?: number;
  current_value?: number;
}

export interface InsightsResult {
  insights: Insight[];
  warnings: Warning[];
  compounding_bonuses: CompoundingBonus[];
}

export interface Warning {
  category: string;
  title: string;
  description: string;
  severity: 'Low' | 'Medium' | 'High';
  suggested_actions: string[];
}

export interface CompoundingBonus {
  category: string;
  title: string;
  description: string;
  effect: string;
  duration_weeks: number;
}

export function generateInsights(prevState: GameState, currentState: GameState): InsightsResult {
  const insights: Insight[] = [];
  const warnings: Warning[] = [];
  const compoundingBonuses: CompoundingBonus[] = [];

  // Generate insights based on state changes
  insights.push(...generateMetricInsights(prevState, currentState));
  insights.push(...generateTrendInsights(currentState));
  insights.push(...generateStrategicInsights(currentState));

  // Generate warnings
  warnings.push(...generateWarnings(currentState));

  // Generate compounding bonuses
  compoundingBonuses.push(...generateCompoundingBonuses(currentState));

  return {
    insights,
    warnings,
    compounding_bonuses: compoundingBonuses,
  };
}

function generateMetricInsights(prevState: GameState, currentState: GameState): Insight[] {
  const insights: Insight[] = [];

  // Revenue insights
  const revenueChange = currentState.mrr - prevState.mrr;
  if (Math.abs(revenueChange) > prevState.mrr * 0.1) {
    if (revenueChange > 0) {
      insights.push({
        category: 'Revenue',
        title: 'Revenue Growth',
        observation: `MRR increased by $${revenueChange.toFixed(0)}`,
        insight: 'Recent actions are driving revenue growth',
        action_suggestion: 'Continue focusing on revenue-generating activities',
        severity: 'Info',
        metric: 'MRR',
        current_value: currentState.mrr,
      });
    } else {
      insights.push({
        category: 'Revenue',
        title: 'Revenue Decline',
        observation: `MRR decreased by $${Math.abs(revenueChange).toFixed(0)}`,
        insight: 'Revenue is declining - investigate churn or pricing issues',
        action_suggestion: 'Run pricing experiments or improve product quality',
        severity: 'Warning',
        metric: 'MRR',
        current_value: currentState.mrr,
      });
    }
  }

  // User growth insights
  const userChange = currentState.wau - prevState.wau;
  if (Math.abs(userChange) > prevState.wau * 0.05) {
    if (userChange > 0) {
      insights.push({
        category: 'Users',
        title: 'User Growth',
        observation: `WAU increased by ${userChange}`,
        insight: 'User acquisition is working well',
        action_suggestion: 'Scale successful acquisition channels',
        severity: 'Info',
        metric: 'WAU',
        current_value: currentState.wau,
      });
    } else {
      insights.push({
        category: 'Users',
        title: 'User Decline',
        observation: `WAU decreased by ${Math.abs(userChange)}`,
        insight: 'Users are leaving - check product quality and competition',
        action_suggestion: 'Investigate churn reasons and improve product',
        severity: 'Warning',
        metric: 'WAU',
        current_value: currentState.wau,
      });
    }
  }

  // Morale insights
  const moraleChange = currentState.morale - prevState.morale;
  if (Math.abs(moraleChange) > 5) {
    if (moraleChange < -5) {
      insights.push({
        category: 'Team',
        title: 'Morale Decline',
        observation: `Team morale dropped by ${Math.abs(moraleChange).toFixed(1)} points`,
        insight: 'Team is becoming demotivated',
        action_suggestion: 'Consider coaching, process improvements, or taking breaks',
        severity: 'Warning',
        metric: 'Morale',
        current_value: currentState.morale,
      });
    } else if (moraleChange > 5) {
      insights.push({
        category: 'Team',
        title: 'Morale Boost',
        observation: `Team morale increased by ${moraleChange.toFixed(1)} points`,
        insight: 'Team is energized and productive',
        action_suggestion: 'Maintain positive momentum',
        severity: 'Info',
        metric: 'Morale',
        current_value: currentState.morale,
      });
    }
  }

  // Tech debt insights
  const debtChange = currentState.tech_debt - prevState.tech_debt;
  if (debtChange > 5) {
    insights.push({
      category: 'Technical',
      title: 'Tech Debt Increase',
      observation: `Technical debt increased by ${debtChange.toFixed(1)} points`,
      insight: 'Code quality is deteriorating',
      action_suggestion: 'Schedule refactoring time or hire more engineers',
      severity: 'Warning',
      metric: 'Tech Debt',
      current_value: currentState.tech_debt,
    });
  }

  return insights;
}

function generateTrendInsights(state: GameState): Insight[] {
  const insights: Insight[] = [];

  // Growth rate trends
  if (state.wau_growth_rate > 15) {
    insights.push({
      category: 'Growth',
      title: 'Strong Growth Momentum',
      observation: `WAU growth rate is ${state.wau_growth_rate.toFixed(1)}%`,
      insight: 'Product has strong market traction',
      action_suggestion: 'Consider scaling operations and hiring',
      severity: 'Info',
    });
  } else if (state.wau_growth_rate < 5) {
    insights.push({
      category: 'Growth',
      title: 'Slow Growth',
      observation: `WAU growth rate is only ${state.wau_growth_rate.toFixed(1)}%`,
      insight: 'Growth has stalled - need to find new acquisition channels',
      action_suggestion: 'Experiment with new marketing channels or product features',
      severity: 'Warning',
    });
  }

  // Cash position
  if (state.bank < state.burn * 3) {
    insights.push({
      category: 'Finance',
      title: 'Cash Runway Concern',
      observation: `Only ${Math.floor(state.bank / state.burn)} months of runway`,
      insight: 'Cash position is precarious',
      action_suggestion: 'Focus on revenue growth or consider fundraising',
      severity: 'Critical',
      metric: 'Bank',
      current_value: state.bank,
    });
  }

  // Reputation trends
  if (state.reputation > 75) {
    insights.push({
      category: 'Reputation',
      title: 'Strong Market Position',
      observation: `Reputation score of ${state.reputation.toFixed(1)}`,
      insight: 'Company has established credibility',
      action_suggestion: 'Leverage reputation for partnerships and hiring',
      severity: 'Info',
    });
  }

  // Compliance risk
  if (state.compliance_risk > 50) {
    insights.push({
      category: 'Compliance',
      title: 'High Compliance Risk',
      observation: `Compliance risk at ${state.compliance_risk.toFixed(1)}%`,
      insight: 'Regulatory issues could threaten the business',
      action_suggestion: 'Prioritize compliance work and legal review',
      severity: 'Critical',
      metric: 'Compliance Risk',
      current_value: state.compliance_risk,
    });
  }

  // NPS trends
  if (state.nps < 0) {
    insights.push({
      category: 'Customer',
      title: 'Poor Customer Satisfaction',
      observation: `NPS is ${state.nps.toFixed(1)}`,
      insight: 'Customers are unhappy with the product',
      action_suggestion: 'Focus on product quality and customer support',
      severity: 'Warning',
      metric: 'NPS',
      current_value: state.nps,
    });
  }

  return insights;
}

function generateStrategicInsights(state: GameState): Insight[] {
  const insights: Insight[] = [];

  // Strategic opportunities based on current state
  if (state.reputation > 70 && state.bank > state.burn * 6) {
    insights.push({
      category: 'Strategy',
      title: 'Fundraising Opportunity',
      observation: 'Strong reputation and healthy cash position',
      insight: 'Market conditions are favorable for fundraising',
      action_suggestion: 'Consider raising capital to accelerate growth',
      severity: 'Info',
    });
  }

  if (state.tech_debt > 40 && state.velocity < 0.8) {
    insights.push({
      category: 'Strategy',
      title: 'Technical Debt Crisis',
      observation: 'High tech debt is severely impacting velocity',
      insight: 'Development speed has slowed significantly',
      action_suggestion: 'Prioritize refactoring and technical debt reduction',
      severity: 'Critical',
    });
  }

  if (state.morale > 80 && state.velocity > 1.2) {
    insights.push({
      category: 'Strategy',
      title: 'High Performance Period',
      observation: 'Team is highly motivated and productive',
      insight: 'This is an optimal time for ambitious projects',
      action_suggestion: 'Tackle complex features or strategic initiatives',
      severity: 'Info',
    });
  }

  if (state.wau > 50000 && state.mrr / state.wau < 10) {
    insights.push({
      category: 'Strategy',
      title: 'Monetization Opportunity',
      observation: `ARPU is only $${(state.mrr / state.wau).toFixed(2)}`,
      insight: 'Large user base with low revenue per user',
      action_suggestion: 'Experiment with pricing tiers or upselling',
      severity: 'Info',
    });
  }

  return insights;
}

function generateWarnings(state: GameState): Warning[] {
  const warnings: Warning[] = [];

  // Critical warnings
  if (state.bank < 0) {
    warnings.push({
      category: 'Finance',
      title: 'Negative Cash Balance',
      description: 'Company has negative cash balance',
      severity: 'High',
      suggested_actions: [
        'Immediately cut costs',
        'Seek emergency funding',
        'Focus on revenue generation',
      ],
    });
  }

  if (state.morale <= 20) {
    warnings.push({
      category: 'Team',
      title: 'Critical Morale Issues',
      description: 'Team morale is critically low',
      severity: 'High',
      suggested_actions: [
        'Address team concerns immediately',
        'Consider leadership changes',
        'Take immediate action to improve work conditions',
      ],
    });
  }

  // Medium warnings
  if (state.tech_debt > 80) {
    warnings.push({
      category: 'Technical',
      title: 'Critical Technical Debt',
      description: 'Technical debt has reached critical levels',
      severity: 'Medium',
      suggested_actions: [
        'Stop all new feature development',
        'Dedicate team to refactoring',
        'Consider hiring senior engineers',
      ],
    });
  }

  if (state.compliance_risk > 70) {
    warnings.push({
      category: 'Compliance',
      title: 'Severe Compliance Risk',
      description: 'Company faces severe regulatory risk',
      severity: 'Medium',
      suggested_actions: [
        'Hire compliance officer',
        'Conduct legal review',
        'Implement compliance processes',
      ],
    });
  }

  // Low warnings
  if (state.churn_rate > 10) {
    warnings.push({
      category: 'Customer',
      title: 'High Churn Rate',
      description: 'Customer churn rate is elevated',
      severity: 'Low',
      suggested_actions: [
        'Improve product quality',
        'Enhance customer support',
        'Run retention experiments',
      ],
    });
  }

  if (state.velocity < 0.5) {
    warnings.push({
      category: 'Product',
      title: 'Low Development Velocity',
      description: 'Team development speed is very low',
      severity: 'Low',
      suggested_actions: [
        'Address technical debt',
        'Improve team processes',
        'Consider hiring additional developers',
      ],
    });
  }

  return warnings;
}

function generateCompoundingBonuses(state: GameState): CompoundingBonus[] {
  const bonuses: CompoundingBonus[] = [];

  // Reputation compounding
  if (state.reputation > 60) {
    bonuses.push({
      category: 'Reputation',
      title: 'Reputation Flywheel',
      description: 'High reputation attracts better talent and customers',
      effect: '+10% hiring success, +5% customer conversion',
      duration_weeks: -1, // Permanent
    });
  }

  // Scale bonuses
  if (state.wau > 10000) {
    bonuses.push({
      category: 'Scale',
      title: 'Network Effects',
      description: 'Large user base creates network effects',
      effect: '+15% organic growth, +5% reputation growth',
      duration_weeks: -1,
    });
  }

  // Product quality bonuses
  if (state.tech_debt < 20) {
    bonuses.push({
      category: 'Quality',
      title: 'Quality Advantage',
      description: 'Low technical debt enables fast, reliable development',
      effect: '+20% development velocity, +10% morale',
      duration_weeks: -1,
    });
  }

  // Team bonuses
  if (state.morale > 70) {
    bonuses.push({
      category: 'Team',
      title: 'High Morale Bonus',
      description: 'Motivated team performs at higher levels',
      effect: '+15% velocity, +10% innovation chance',
      duration_weeks: -1,
    });
  }

  // Market position bonuses
  if (state.momentum > 60) {
    bonuses.push({
      category: 'Momentum',
      title: 'Momentum Advantage',
      description: 'Strong momentum attracts investors and talent',
      effect: '+20% fundraising success, +10% hiring success',
      duration_weeks: 12,
    });
  }

  return bonuses;
}